client/components/ChallengesList.js

```diff
@@ -1,26 +1,39 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
 import { connect } from 'react-redux';
-import { fetchChallenges } from '../store';
+import { fetchChallenges, fetchSearchChallenges, fetchFilterChallenges } from '../store';
 import { convertBufferToImgSrc } from '../utils';
+import { Search } from './Search';
+import { Filter } from './Filter';
 
 const mapDispatchToProps = dispatch => ({
   fetchChallenges: () => dispatch(fetchChallenges()),
+  fetchSearchChallenges: term => dispatch(fetchSearchChallenges(term)),
+  fetchFilterChallenges: difficulty => dispatch(fetchFilterChallenges(difficulty)),
 });
 
 const mapStateToProps = ({ challenges }) => ({ challenges });
 
-const component = ({ challenges, fetchChallenges }) => {
+const component = props => {
+  const { searchTerm, difficulty } = props.match.params;
   useEffect(() => {
-    fetchChallenges();
-  }, []);
+    if (!searchTerm && !difficulty) {
```

Comment:
Think about inverting this if...
`if (searchTerm && difficulty) {
    // search by both
} else if (searchTerm) {
    // search by the searchTerm
} else if (difficulty) {
    // search by the difficulty
} else {
    // search normally
}`